tacs/constitutive.pyx

```diff
@@ -216,9 +218,29 @@ cdef class MaterialProperties:
         cdef TacsScalar kappa3 = 230.0
 
         ISOTROPIC_KEYS = ["rho", "E", "nu", "G", "ys", "alpha", "specific_heat", "kappa"]
+        ORTHOTROPIC_KEYS = ["E1", "E2", "E3",
+                            "nu12", "nu13", "nu23",
+                            "G12", "G13", "G23",
+                            "T1", "T2", "T3",
+                            "C1", "C2", "C3",
+                            "S12", "S13", "S23",
+                            "Xt", "Yt", "Xc", "Yc", "S",
+                            "alpha1", "alpha2", "alpha3",
+                            "kappa1", "kappa2", "kappa3"]
+        ALL_KEYS = ISOTROPIC_KEYS + ORTHOTROPIC_KEYS
+
+        # Check for any invalid/misspelled inputs
+        for input_key in kwargs:
+          if input_key not in ALL_KEYS:
+              warnings.warn(
```

Comment:
I would prefer this result in an exception than a warning, it's easy to miss a warning amongst a bunch of other output. If a material property I thought I was setting is not getting set then whatever analysis I'm running is pretty much guaranteed to produce a useless result anyway.

Reply:
I don't disagree with this. I do worry that it would be a breaking change (though an easy one to fix) and could lead to a lot of disruption downstream. 

I think if we were going to go the error route, we should just refactor to get rid of kwargs and explicitly define expected arguments in the function handle anyways, which would be a bigger change.

Reply:
@A-CGray I'm fine with leaving it an error